src/Compilers/CSharp/Portable/Binder/SubsumptionDiagnosticBuilder.cs

```diff
@@ -129,6 +129,8 @@ private ErrorCode CheckSubsumed(BoundPattern pattern, DecisionTree decisionTree,
                         // the input can no longer be null. In that case a null pattern is subsumed.
                         if (isNull && !inputCouldBeNull)
                         {
+                            // Note: we do not have any test covering this. Is it reachable?
```

Comment:
I agree with @AlekseyTs regarding removing code that we think is unreachable and is not tested. If we'd rather not delete the few lines of code here and below, consider commenting it out and `throw ExceptionUtilities.Unreachable;` instead.